### ISSUE TITLE
Support `split column` for `table` input

### ISSUE THREAD

**Issue by Dimagog** [nushell/nushell#10748]
### Related problem

Quite often parsed tables still have columns that require additional parsing into sub-columns.
So, 3 suggestions:

1) Today `split column` could be used on a `string` and on a `list<string>`.
It would be useful to also support a `table` as an input. In other words I'd like to be able to `split column` for an existing table into multiple columns.

2) Additionally add a `--keep` switch to keep the original column (by default delete it).

3) Support `--keep` switch for a `list<string>` input, preserving the original input into its own column.

### 1) Table input without `--keep` switch
The first `rest` param (`a` below) is considered to be a name of existing column to be split, the original column (`a` below) is **removed**
```
~> [[a]; ["1,2"] ["3,4"]] | split column "," a b c
```
to return the same as the following does today:
```
~> [[a]; ["1,2"] ["3,4"]] | each { $in | get a | split column "," b c | get 0}
╭───┬───┬───╮
│ # │ b │ c │
├───┼───┼───┤
│ 0 │ 1 │ 2 │
│ 1 │ 3 │ 4 │
╰───┴───┴───╯
```
### 2) Table input with `--keep` switch
The first `rest` param (`a` below) is considered to be a name of existing column to be split, the original column (`a` below) is **preserved**
```
~> [[a]; ["1,2"] ["3,4"]] | split column --keep "," a b c
```
to return the same as the following does today:
```
[[a]; ["1,2"] ["3,4"]] | each { $in | merge ($in | get a | split column "," b c | get 0 )}
╭───┬─────┬───┬───╮
│ # │  a  │ b │ c │
├───┼─────┼───┼───┤
│ 0 │ 1,2 │ 1 │ 2 │
│ 1 │ 3,4 │ 3 │ 4 │
╰───┴─────┴───┴───╯
```
### 3) `List<string>` input with `--keep` switch
```
["1,2" "3,4"] | split column --keep "," a b c
```
to return the same as the following does today, preserving original data into `a`:
```
["1,2" "3,4"] | each {|$it| {a: $it} | merge ($it | split column "," b c| get 0)}
╭───┬─────┬───┬───╮
│ # │  a  │ b │ c │
├───┼─────┼───┼───┤
│ 0 │ 1,2 │ 1 │ 2 │
│ 1 │ 3,4 │ 3 │ 4 │
╰───┴─────┴───┴───╯
```

**Comment by Dimagog** [comment]
Could be grouped with #10747

**Comment by amtoine** [comment]
> Could be grouped with #10747
>
do you mean this issue is a duplicate of #10747?

**Comment by Dimagog** [comment]
No, I mean both changes would touch the same code, and it could be more efficient to work on both together. But they are different. #10747 is a bug, and this one is an improvement suggestion.

**Comment by amtoine** [comment]
@Dimagog 
ooh okey, gotcha :ok_hand:

**Comment by weirdan** [comment]
There are clearer alternatives to those proposed by the original reporter:

```nushell
# [[a]; ["1,2"] ["3,4"]] | split column "," a b c
[[a]; ["1,2"] ["3,4"]] | update a { split column "," b c } | flatten --all a


# [[a]; ["1,2"] ["3,4"]] | split column --keep "," a b c
[[a]; ["1,2"] ["3,4"]] | insert new { get a | split column "," b c } | flatten --all new
```

The requested feature would still be useful though.
